nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/eventhub/ConsumeAzureEventHub.java

```diff
@@ -486,11 +563,36 @@ protected String getTransitUri(final PartitionContext partitionContext) {
             if (readerFactory == null || writerFactory == null) {
                 writeFlowFiles(eventBatchContext, session, stopWatch);
             } else {
-                writeRecords(eventBatchContext, session, stopWatch);
+                writeRecords(eventBatchContext, session, readerFactory, writerFactory, stopWatch);
             }
 
-            // Commit ProcessSession and then update Azure Event Hubs checkpoint status
-            session.commitAsync(eventBatchContext::updateCheckpoint);
+            // As a special case, when the checkpoint strategy is component state,
+            // we reuse the current session.
+            if (checkpointStrategy == CheckpointStrategy.COMPONENT_STATE) {
+                eventBatchContext = new EventBatchContext(
+                        eventBatchContext.getPartitionContext(),
+                        eventBatchContext.getEvents(),
+                        new ComponentStateCheckpointStore(
+                                identifier,
+                                new ComponentStateCheckpointStore.State() {
+                                    public StateMap getState() throws IOException {
+                                        return session.getState(Scope.CLUSTER);
+                                    }
+
+                                    public boolean replaceState(StateMap oldValue, Map<String, String> newValue) {
+                                        return false;
+                                    }
+                                }
+                        ),
```

Comment:
The inner anonymous class results in multiple levels of nesting that make this hard to follow. Recommend breaking it out to an explicit class.